src/web/app/pages-instructor/instructor-session-base-page.component.ts

```diff
@@ -2,35 +2,35 @@ import { Router } from '@angular/router';
 import { NgbModal, NgbModalRef } from '@ng-bootstrap/ng-bootstrap';
 import { saveAs } from 'file-saver';
 import { from, Observable, of } from 'rxjs';
-import { concatMap, finalize, last, switchMap } from 'rxjs/operators';
+import { catchError, concatMap, finalize, last, switchMap } from 'rxjs/operators';
 import { FeedbackQuestionsService } from '../../services/feedback-questions.service';
 import { FeedbackSessionsService } from '../../services/feedback-sessions.service';
 import { InstructorService } from '../../services/instructor.service';
 import { NavigationService } from '../../services/navigation.service';
 import { StatusMessageService } from '../../services/status-message.service';
 import { TableComparatorService } from '../../services/table-comparator.service';
 import {
-    FeedbackQuestion,
-    FeedbackQuestions,
-    FeedbackSession,
-    FeedbackSessionStats,
-    InstructorPrivilege,
+  FeedbackQuestion,
+  FeedbackQuestions,
+  FeedbackSession,
+  FeedbackSessionStats,
+  InstructorPrivilege,
 } from '../../types/api-output';
 import { Intent } from '../../types/api-request';
 import { DEFAULT_NUMBER_OF_RETRY_ATTEMPTS } from '../../types/default-retry-attempts';
 import { SortBy, SortOrder } from '../../types/sort-properties';
+import { CopySessionModalResult } from '../components/copy-session-modal/copy-session-modal-model';
 import { ErrorReportComponent } from '../components/error-report/error-report.component';
-import {
-    CopySessionResult,
-    SessionsTableRowModel,
-} from '../components/sessions-table/sessions-table-model';
+import { CopySessionResult, SessionsTableRowModel } from '../components/sessions-table/sessions-table-model';
 import { ErrorMessageOutput } from '../error-message-output';
 
 /**
  * The base page for session related page.
  */
 export abstract class InstructorSessionBasePageComponent {
 
+  protected failedToCopySessions: Record<string, string> = {}; // Map of failed session copy to error message
```

Comment:
This should not be part of the component, but admittedly I can't really think of a better way to put this that still achieves what we want.